third_party/move/move-core/types/src/effects.rs

```diff
@@ -276,25 +287,31 @@ impl ChangeSet {
         })
     }
 
-    pub fn modules(&self) -> impl Iterator<Item = (AccountAddress, &Identifier, Op<&[u8]>)> {
+    pub fn modules(&self) -> impl Iterator<Item = (AccountAddress, &Identifier, Op<&M>)> {
         self.accounts.iter().flat_map(|(addr, account)| {
             let addr = *addr;
             account
                 .modules
                 .iter()
-                .map(move |(module_name, op)| (addr, module_name, op.as_ref().map(|v| v.as_ref())))
+                .map(move |(module_name, op)| (addr, module_name, op.as_ref()))
         })
     }
 
-    pub fn resources(&self) -> impl Iterator<Item = (AccountAddress, &StructTag, Op<&[u8]>)> {
+    pub fn resources(&self) -> impl Iterator<Item = (AccountAddress, &StructTag, Op<&R>)> {
         self.accounts.iter().flat_map(|(addr, account)| {
             let addr = *addr;
             account
                 .resources
                 .iter()
-                .map(move |(struct_tag, op)| (addr, struct_tag, op.as_ref().map(|v| v.as_ref())))
+                .map(move |(struct_tag, op)| (addr, struct_tag, op.as_ref()))
         })
     }
 }
 
+// These aliases are necessary because AccountChangeSet and ChangeSet were not
+// generic before. In order to minimise the code changes we alias new generic
+// types.
+pub type AccountChangeSet = AccountChanges<Vec<u8>, Vec<u8>>;
+pub type ChangeSet = Changes<Vec<u8>, Vec<u8>>;
```

Comment:
Where is this still used if we aren't serializing?

Reply:
These are used in tests, in `async-vm`, and are be produced by calling `session.finish()` (basically no old behaviour has changed, only a small overhead of forming new sets)